### ISSUE TITLE
Not initiated when opening file directly

### ISSUE THREAD

**Issue by d-miketa** [lewis6991/gitsigns.nvim#315]
I ran across the exact behaviour described in https://github.com/lewis6991/gitsigns.nvim/issues/18, but with a Lua config. Opening a file like `nvim path/to/file` doesn't load gitsigns but `:e path/to/file` does, as does opening a file directly and then reloading it with `:e`.

The relevant bits in my config are

```        
use {'lewis6991/gitsigns.nvim',
         requires = {'nvim-lua/plenary.nvim'},
         config = function() require'gitsigns'.setup({ keymaps = { } }) end
}
```

**Comment by lewis6991** [comment]
Please fill out the issue template.

Without providing real instructions on how to reproduce this or making sure the issue persists in a minimal config, there is very little anyone can help with.

90% of the time, these kinds of issues are just because of bad configuration, weird nvim install, or a platform issue.

**Comment by d-miketa** [comment]
Sure.

`gitsigns-debug.lua`:
```
vim.o.termguicolors = true 

require('gitsigns').setup {
    debug_mode = true
}
```

0. Make sure `gitsigns` is available for loading.
1. Run `nvim -u gitsigns-debug.lua file/under/git/with/changes`. Observe that `gitsigns` isn't running.
2. Execute `:e`; `gitsigns` shows up.
3. Now try `nvim -u gitsigns-debug.lua` and `:e file/under/git/with/changes`, it works.

platform: macOS Big Sur 11.5.2
nvim: v0.6.0-dev+1637-g2ae9ff128 (reproduced with 0.5.0)

**Comment by lewis6991** [comment]
And debug messages?

**Comment by d-miketa** [comment]
Sorry, missed that one.

Step 1:
<img width="1148" alt="image" src="https://user-images.githubusercontent.com/320321/130156175-4ec17f32-8fa8-4b2d-b300-e117ab27912a.png">

Step 2:
<img width="1189" alt="image" src="https://user-images.githubusercontent.com/320321/130156230-929e59a2-0c48-425f-9f1c-0996dfa697f1.png">

**Comment by lewis6991** [comment]
Can you provide the output of the cache? Details in the issue template.

It looks like the call to `git ls-files` isn't returning the right data on the first call, but is on the second call.

Also what git version are you running?

**Comment by d-miketa** [comment]
I just tried to reproduce the issue to fix debug but can't. Everything seems to be working well now. I did purge my `nvim/plugin` and `nvim/lua` directories to prevent as many things from loading as possible, and following the steps described above no longer results in an error. Could be there was a recent update to neovim or this plugin that fixed it? I have `git==2.33.0` by the way, installed via Homebrew; perhaps their formula was recently updated to this version.

I'm closing the issue as it seems to have resolved itself. Thanks for your patience and a great plugin!

**Comment by lewis6991** [comment]
Ok no problem. I think there is still an underlying issue here, as it is similar to other issues in the past. Specifically, the first call of `git ls-files` not being correct. Maybe some latency issue?

However, if it's something that can't be reproduced, then it's probably best to close the issue for now.
